### PR TITLE
chore: Update hedera-services to rename develop as main

### LINKED ISSUE #16809: Rename `develop` to `main`
Performs re-name on `hedera-services` default branch to change it from `develop` to `main`

### PR DESCRIPTION
**Pull request by rbarkerSL**
**Description**:

Updates references to `develop` to point to the `main` branch in hedera-services.

**Related issue(s)**:

Fixes #16809 

**Notes**:
This PR is expected to merge on 12/20/2024 as part of the change of the default branch name

### CODE REVIEW

File: gradle/plugins/src/main/kotlin/com.hedera.gradle.spotless.gradle.kts [hunk 1]
```diff
@@ -21,7 +21,7 @@ spotless {
     isEnforceCheck = false
 
     // optional: limit format enforcement to just the files changed by this feature branch
-    ratchetFrom("origin/develop")
+    ratchetFrom("origin/main")
```
Comment (jjohannes) [inline gradle/plugins/src/main/kotlin/com.hedera.gradle.spotless.gradle.kts:24]: The file modified here no longer exists. For this change, you now need to delete this file:
https://github.com/hashgraph/hedera-services/blob/develop/gradle/development-branch.txt

If `development-branch.txt` does not exist, we defaults to `main`.

**Review by mishomihov00 — DISMISSED** [review]

**Review by andrewb1269hg — APPROVED** [review]

File: gradle/development-branch.txt
Comment (andrewb1269hg) [inline gradle/development-branch.txt:1]: I see @jjohannes added this file in PR #15282 . Do we need to leave this file in, update it, or remove it? @rbarkerSL

Comment (jjohannes) [reply gradle/development-branch.txt:1]: The file can and should be removed. 'main' is the default if the file does not exist. (Information is used by Spotless to decide which files to check.)

File: docs/maintainers-guide.md [hunk 2]
```diff
@@ -92,35 +90,34 @@ with 0.30 milestone on it.
...
 ![labels-on-issue](./assets/labels-on-issue.png)
 
-### Release Engineering Responsibilities
+### DevOps-CI Responsibilities
 
 The release engineering team will handle the following:
 
-- Create a release branch from `develop` branch at the end of first sprint in the release cycle
-- Will merge the release branch for current deploying release into `main`
+- Create a release branch from `main` branch at the end of first sprint in the release cycle
```
Comment (nathanklick) [inline docs/maintainers-guide.md:97]: This needs to be fixed

File: docs/maintainers-guide.md [hunk 3]
```diff
@@ -92,35 +90,34 @@ with 0.30 milestone on it.
 
 ![labels-on-issue](./assets/labels-on-issue.png)
 
-### Release Engineering Responsibilities
+### DevOps-CI Responsibilities
 
 The release engineering team will handle the following:
```
Comment (nathanklick) [inline docs/maintainers-guide.md:95]:
```suggestion
The DevOps CI team will handle the following:
```

**Pull request merged**